pkg/trojan/message.go

```diff
@@ -0,0 +1,283 @@
+// Copyright 2020 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package trojan
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"hash"
+	"math/big"
+
+	"github.com/ethersphere/bee/pkg/crypto"
+	"github.com/ethersphere/bee/pkg/swarm"
+	bmtlegacy "github.com/ethersphere/bmt/legacy"
+	"golang.org/x/crypto/sha3"
+)
+
+// Topic is an alias for a 32 byte fixed-size array which contains an encoding of a message topic
+type Topic [32]byte
+
+// Target is an alias for an address which can be mined to construct a trojan message
+type Target []byte
```

Comment:
What does it bring to type this separately, instead of just using `swarm.Address`?

Reply:
targets are partial addresses. [eventually](https://github.com/ethersphere/swarm/issues/2156), they will be bit vectors

Reply:
I think it is important to document that the target length is dependent on the network size. I believe that right now, target is intended to be given as a parameter, right? So, it'll be the responsibility of the client to give the correct parameter size. 

Well, that's fine. But it should be documented somewhere how Target relates to the address and what the length of the Target should be. 

If this documentation is elsewhere, please ignore the comment!

Reply:
> I think it is important to document that the target length is dependent on the network size

proposal:

```
type Target swarm.Address
```

Then `Target` can add methods on top of that for anything opinionated. It does however represent an actual swarm address, as in there is nothing in the `pkg/swam/swarm.go` definition/implementation of `Address` that says anything about size.

Reply:
it is a good proposal, but i don't know that we should invest time on it if we plan to implement this through bit vectors anyway.

thoughts?

Reply:
Agree. In my opinion, some documentation is minimally needed.